app/src/lib.rs

```diff
@@ -275,25 +275,22 @@ impl PortalApp {
         while let Ok(response) = rx.next().await.ok_or(AppError::ListenerDisconnected)? {
             log::debug!("Received auth challenge: {:?}", response);
 
-            let result = evt.on_auth_challenge(response.clone()).await?;
-            log::debug!("Auth challenge callback result: {:?}", result);
+            let approved = evt.on_auth_challenge(response.clone()).await?;
```

Comment:
You should update `on_auth_challenge` to return an `AuthResponseStatus` instead of the boolean. So it contains the token and granted permissions